### ISSUE TITLE
PG16Beta3 Support - Regression tests sanity

### ISSUE THREAD

**Issue by naisila** [citusdata/citus#7017]
This issue encapsulates what remains to be fixed in #6952

The following should be fixed for successful compilation:
- [x] Find a way to retrieve selectedCols for child relation
turns out that it was related to error in different order in Parallel Append, seen in `columnar_partitioning`
and also new Join Filter node on top, causing diffs, seen in `columnar_chunk_filtering`

And the rest:
- [x] Fix `ERROR:  disabling the first worker node in the metadata is not allowed` in `isolation_add_remove_node`
note: **can't reproduce in my local** ❗❗
- [x] unexpected local crash in `sqlancer_failures`, test passes in CI
```SQL
CREATE TABLE distributed_table(user_id int, item_id int, buy_count int);
SELECT create_distributed_table('distributed_table', 'user_id');
CREATE TABLE reference_table(id int, it_name varchar(25), k_no int);
SELECT create_reference_table('reference_table');

SELECT
   count(*)
FROM
   (
    SELECT a.* FROM distributed_table a JOIN distributed_table b USING (user_id)
   ) AS bar
JOIN
   (
      SELECT a.* FROM distributed_table a
      LEFT JOIN (SELECT * FROM reference_table OFFSET 0) b ON (true)
      RIGHT JOIN (SELECT * FROM reference_table OFFSET 0) c ON (true)
   ) AS unsupported_join
ON (true);

server closed the connection unexpectedly
        This probably means the server terminated abnormally
        before or while processing the request.
```
- [x] Wal2json is needed for cdc tests, currently disabled them in CI, need to figure out a way around this 
- [x] 🟠 Fix `ERROR:  "table mx_ddl_table2" has dependency on unsupported object "access method heap2"`, seen in `pg15`
- [x] 🟠 Not getting `ERROR:  cannot push down this subquery`, seen in `local_table_join`
- [x] 🟡PG16 explain diffs: sort instead of index scan, seen in `multi_complex_count_distinct`
- [x] 🟡PG16 explain diffs: extra entry in Sort Key node, seen in `columnar_paths`, `multi_explain`, `multi_orderby_limit_pushdown`, `multi_complex_count_distinct`, `multi_select_distinct`
Could be the relevant PG commit https://github.com/postgres/postgres/commit/1349d2790bf48a4de072931c722f39337e72055e
- [x] 🟡PG16 explain diffs: merge join instead of hash join, seen in `multi_explain`
- [x] 🟡PG16 explain diffs: extra SORT node, seen in `multi_explain`, `multi_complex_count_distinct`
- [x] Fix `ERROR:  permission denied to cancel query`, seen in `global_cancel`
- [x] Not getting expected `ERROR:  must be owner of table test_table` in `isolation_multiuser_locking` PASS IN BETA3
- [x] Implement grant-level control of role inheritance, seen in `create_role_propagation`
Partially done - only supporting WITH ADMIN OPTION properly as we already did. Now regression tests pass. More work to be done to support the whole GRANT changes.
- [x] [FIX THIS disable the problematic index scans in columnar metadata tables](https://github.com/citusdata/citus/pull/6952/commits/8f1b88ba9d3e2211983bb84d384fd7560f137c01)
- [x] [FIX THIS AdjustPartitioningForDistributedPlanning crashes!](https://github.com/citusdata/citus/pull/6952/commits/b618f11a47879e9b345671191073e1c99d3c1a30)
- [x] [FIX THIS SOmehow views shows up as subqueries with relationId set](https://github.com/citusdata/citus/pull/6952/commits/d22528e807a1d58e246675c2c3971ca5dfbd7430)
Caused by https://github.com/postgres/postgres/commit/47bb9db75996232ea71fc1e1888ffb0e70579b54 - This was failing in beta1 but got fixed in beta2 by https://github.com/postgres/postgres/commit/0f8cfaf8921fed35f0b92d918ce95eec7b46ff05
- [x] [FIX THIS related to outer join checks, root->simple_rel_array](https://github.com/citusdata/citus/pull/6952/commits/7be9a31aff8e29aa8f2ecc5256ccbdc355ed1632)
- [x] 14-16 and 15-16 pg upgrade tests - there is an error in `Performing Upgrade`
was related to any_value
- [x] Max mem usage changed and it's no longer less than 8MB, seen in `columnar_memory`
It's still less than 9MB. TopMemoryContext simply has more children.
- [x] Fix `ERROR:  duplicate key value violates unique constraint` in `columnar_write_concurrency_index`
- [x] Extra `DEBUG:  pathlist hook for columnar table am`, seen in `drop_column_partitioned_table`
- [x] Not getting expected `ERROR:  correlated subqueries are not supported when the FROM clause contains a CTE or subquery`, seen in `multi_subquery_in_where_reference_clause`
- [x] Fix `ERROR:  relation "rule_table_1" cannot have ON SELECT rules`, seen in `undistribute_table`
this fails on plain PG16 whereas it succeeds in plain PG15, so not related to Citus
- [x] Fix `ERROR:  "rule_table_1" is not a view`, seen in `undistribute_table`
this fails on plain PG16 whereas it succeeds in plain PG15, so not related to Citus
- [x] Not getting expected `ERROR:  cannot alter table because an extension depends on it`, seen in `undistribute_table`
- [x] Different `pa_compare_tables` output, seen in `merge`
- [x] Fix `ERROR:  EXPLAIN ANALYZE is currently not supported for MERGE INTO ... commands with repartitioning`, seen in `merge`
- [x] Fix `DEBUG:  MERGE command is only supported when all distributed tables are co-located and joined on their distribution columns`, seen in `merge`
This works if we don't AdjustPartitioning. Fixed by https://github.com/citusdata/citus/commit/4ae3982d14f54e3c50f5e2d9b1d840ca412977cc
- [x] Fix `ERROR:  invalid perminfoindex 2 in RTE with relid 34041`, seen in `merge_repartition1`, `merge_partition_tables`
- [x] Not getting expected `ERROR:  cannot perform select on a distributed table and modify a reference table`, seen in `multi_modifications`
- [x] Not getting expected `ERROR:  relation postgres_table is not distributed`, seen in `local_table_join`
- [x] Fix `ERROR:  unsupported join alias expression`, seen in `sqlancer_failures`
- [x] Shard names not hidden, seen in `multi_mx_hide_shard_names`
- [x] Fix `ERROR:  cannot pushdown the subquery since not all subqueries in the UNION have the partition column in the same position`, seen in `union_pushdown`
- [x] Fix `SELECT NULL::integer AS "dummy-1" FROM local_table_join.local_partitioned_table`, seen in `local_table_join`
- [x] 🟦 Fix `ERROR:  unrecognized token: "0"`, seen in `upgrade_basic_after`, `upgrade_ref2ref_after`, `upgrade_type_after`, `upgrade_distributed_function_after`
- [x] 🟤 Fix `ERROR:  unrecognized configuration parameter "lc_collate"`, seen in `multi_mx_create_table`, `multi_schema_support`
- [x] 🟤 Fix `ERROR:  collation "english" for encoding "UTF8" does not exist`, seen in `multi_mx_create_table`, `multi_schema_support`
- [x] 🟤 Fix `ERROR:  syntax error at or near "0"`, seen in `multi_mx_create_table`
- [x] 🟤 Fix different shard id, seen in `create_distributed_table_concurrently`
- [x] 🟤 Error changed from `permission denied for index pg_toast_1260_index` to `must be owner of index pg_toast_1260_index`, seen in `create_index` PASS IN BETA3
- [x] 🟣Fix `complex joins are only supported when all distributed tables are co-located and joined on their distribution columns`, seen in `multi_insert_select`
- [x] 🟣Different count output, seen in `recurring_outer_join`
- [x] 🟡Changed error output `Either provide the file, use the system's trusted roots with sslrootcert=system,`, seen in `multi_move_mx`
- [x] 🟡Fix `ERROR:  unrecognized VACUUM option "process_main"`, seen in `vacuum`, `stats`
- [x] 🟡Notice changed `NOTICE:  backend type switched to: client backend`, seen in `multi_mx_hide_shard_names`
- [x] 🟡Fix `any_value`, seen in `multi_extension`
- [x] 🟡Extra column entry `varnullingrels` in PG16, seen in `single_shard_table_udfs`, `multi_metadata_sync`, `citus_local_tables`, `start_stop_metadata_sync`, `citus_local_tables_mx`
- [x] 🟡Output order change `äbc` vs `data`, seen in `pg13`
- [x] 🟡Extra `NOTICE:  using standard form "und-u-ks-level2" for locale "@colStrength=secondary"`, seen in `pg12`
- [x] 🟡Hint changed to detail `There is an entry for table "j1_tbl", but it cannot be referenced from this part of the query.`, seen in `pg14`, `pgmerge`
- [x] 🟡`NOTICE:  issuing CREATE SUBSCRIPTION` now includes `password_required`, seen in `pg14`, `cpu_priority`
- [x] 🟡PG16 explain diffs: different order in Parallel Append, seen in `columnar_partitioning`
it was related to wrong numberOfColumnsRead value for child relations, fixed.
- [x] 🟡PG16 explain diffs: different order in filters, `id = r1.id1` vs `r1.id1 = id`, seen in `columnar_chunk_filtering`, `multi_hash_pruning`
- [x] 🟡PG16 explain diffs: new Join Filter node on top, causing diffs, seen in `columnar_chunk_filtering`
it was related to wrong numberOfColumnsRead value for child relations, fixed.
- [x] 🟡PG16 explain diffs: missing entry in Group Key node, seen in `multi_subquery`, `multi_having_pushdown`, `insert_select_repartition`
- [x] 🟡Fix `ERROR:  must be owner of index maintain_test_a_idx`, seen in `privileges` PASS IN BETA3
- [x] 🟡PG16 showing unqualified table columns in debugs, seen in `non_colocated_subquery_joins`, `multi_view`, `undistribute_table`, `local_dist_join_mixed`, `recurring_outer_join`, `replicate_reference_tables_to_coordinator`, `view_propagation`, `citus_local_tables_mx`

------------------------------------

🟣 could be related to #6895 
🟡 should be simple to fix
🟤 extra failures from PG16Beta2
🟠 extra failures from PG16Beta3
🟦 possibly related to varnullingrels

**Comment by onderkalaci** [comment]
What about pg-vanilla tests? Please check with @aykut-bozkurt on that if you are not familiar with

**Comment by naisila** [comment]
@onderkalaci 
> What about pg-vanilla tests?

They are already covered by `test-16_check-vanilla` right? I included issues from there too. They are trivial.
https://app.circleci.com/jobs/github/citusdata/citus/1172842

**Comment by tejeswarm** [comment]
[HPE New features in PG-16](https://h50146.www5.hpe.com/products/software/oe/linux/mainstream/support/lcc/pdf/PostgreSQL16Beta1_New_Features_en_20230528_1.pdf)
